scripts/run_mteb_german.py

```diff
@@ -0,0 +1,58 @@
+import logging
+from typing import List, Optional
+
+import mteb.abstasks.AbsTask
+from mteb import MTEB
+
+logging.basicConfig(level=logging.INFO)
+
+logger = logging.getLogger("main")
+
+
+def _get_preferred_split(available_splits):
+    order = ["test", "valid", "validation", "dev"]
+    for possibility in order:
+        if possibility in available_splits:
+            return possibility
+    return available_splits[0]
+
+
+def evaluate(
+    model,
+    model_name: str,
+    task_list: Optional[List[str]] = None,
+    task_langs: Optional[List[str]] = None,
+):
+    task_list = task_list or TASK_LIST
+    task_langs = task_langs or ["en"]
+    for task in task_list:
+        logger.info(f"Running task: {task}")
+        cat_classes = [cls for cls in mteb.abstasks.AbsTask.__subclasses__()]
+        task_instances = [
+            task_cls()
+            for cat_cls in cat_classes
+            for task_cls in cat_cls.__subclasses__()
+            if cat_cls.__name__.startswith("AbsTask")
+        ]
+        task_instances = [
+            tsk for tsk in task_instances if tsk.description['name'] == task
+        ]
+        assert len(task_instances) == 1, task_instances
+        eval_splits = [
+            _get_preferred_split(task_instances[0].description['eval_splits'])
+        ]
+        evaluation = MTEB(
+            tasks=[task], task_langs=task_langs
+        )
+        evaluation.run(
+            model, output_folder=f"results/{model_name}", eval_splits=eval_splits
+        )
+
+if __name__ == '__main__':
+    from sentence_transformers import SentenceTransformer
+
+    TASK_LIST = ["MIRACL", "GermanDPR", "PawsX", "GermanSTSBenchmark", "XMarket", "GerDaLIR", "WikiCLIR"]
+    MODELS = ['intfloat/multilingual-e5-base', 'intfloat/multilingual-e5-large', 'T-Systems-onsite/cross-en-de-roberta-sentence-transformer', 'sentence-transformers/distiluse-base-multilingual-cased-v2']
+    for model_name in MODELS:
+        model = SentenceTransformer(model_name, device='cuda')
```

Comment:
This automatically limits the `max_seq_length` to 512. If this is desired, then I think the MTEB scores we publish should also result from the same `max_seq_length` of 512 and not 8k.

Reply:
The `sentence-transformers/distiluse-base-multilingual-cased-v2` model actually uses a sequence length of 128. I'm not sure how large the positional embeddings even are for these models.